File: 278. First Bad Version/main.cpp
// Forward declaration of isBadVersion API.
bool isBadVersion(int version);

class Solution {
public:
  int firstBadVersion(int n) {
    // 事实上，在最后一般需要判断。但这题确定target存在。
    int left = 1, right = n, mid;
    while (left < right) {
      mid = (right - left) / 2 + left;
      if (isBadVersion(mid)) {
        right = mid;
      } else {
        left = mid + 1;
      }
    }
    return left;
  }
};

File: 278. First Bad Version/main.py
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return a bool
# def isBadVersion(version):

class Solution:
    def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        left,right = 0,n
        mid = 0
        while left<right:
            mid=(left+right)//2
            if isBadVersion(mid):
                right = mid
            else:
                left = mid + 1

        # 事实上，在最后一般需要判断。但这题确定target存在。
        return left

